File: 이건희/level2/의상.py
def solution(clothes):
    result = 1
    cnt = {}
    
    for i in clothes:
        cloth_type = i[1]
        if cloth_type in cnt:
            cnt[cloth_type] += 1
        else:
            cnt[cloth_type] = 1
            
    for i in cnt:
        result *= cnt[i] + 1
        
    return result - 1

File: 이훈일/Programmers/Level2/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        for(int i = 0; i < clothes.length; i++) {
            String cloth_kind = clothes[i][1];
            if(map.containsKey(cloth_kind)) {
                map.put(cloth_kind, map.get(cloth_kind) + 1);
            } else {
                map.put(cloth_kind, 1);
            }
        }
        
        int sum = 1;
        for(String key : map.keySet()) {
            sum *= map.get(key) + 1;
        }
        
        return sum - 1;
    }
}
